PHPCompatibility/Tests/Syntax/NewConstantDereferencingUnitTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * PHPCompatibility, an external standard for PHP_CodeSniffer.
+ *
+ * @package   PHPCompatibility
+ * @copyright 2012-2021 PHPCompatibility Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCompatibility/PHPCompatibility
+ */
+
+namespace PHPCompatibility\Tests\Syntax;
+
+use PHPCompatibility\Tests\BaseSniffTest;
+
+/**
+ * Test the NewArrayStringDereferencing sniff.
+ *
+ * @group newArrayStringDereferencing
+ * @group syntax
+ *
+ * @covers \PHPCompatibility\Sniffs\Syntax\NewArrayStringDereferencingSniff
+ *
+ * @since 10.0.0
+ */
+class NewConstantDereferencingUnitTest extends BaseSniffTest
+{
+
+    /**
+     * Test for errors related to array dereferencing of constants in
+     * PHP 5.5 and below.
+     *
+     * @dataProvider dataArrayDereferencing
+     *
+     * @param int $line The line number.
+     *
+     * @return void
+     */
+    public function testArrayDereferencing($line)
+    {
+        $file = $this->sniffFile(__FILE__, '5.5');
+        $this->assertError($file, $line, 'Array dereferencing of constants is not present in PHP version 5.5 or earlier');
+    }
```

Comment:
This should also contain an assertion to test that these don't throw an error when the `testVersion` is `5.6`.